src/citrine/resources/predictor.py

```diff
@@ -87,16 +87,20 @@ def auto_configure(self, *, training_data: DataSource, pattern="PLAIN") -> Predi
         training_data: DataSource
             The data to configure the predictor to model.
         pattern: str
-            The predictor pattern to use, either "PLAIN" or "FORMULATION"
+            The predictor pattern to use, either "PLAIN", "FORMULATION", or "INFER".
+            The "INFER" pattern auto-detects whether the `DataSource` is formulations data or not.
+            If it is, then a formulation predictor is created.
+            If not, then a plain predictor is created.
 
         Returns
         -------
         Predictor
             Automatically configured predictor for the training data
 
         """
-        if pattern not in {"FORMULATION", "PLAIN"}:
-            msg = "Called with pattern: {}.  Expected 'FORMULATION' or 'PLAIN'.".format(pattern)
+        if pattern not in {"FORMULATION", "PLAIN", "INFER"}:
```

Comment:
Why are you not creating an enumeration here?  That would solve the user problem of guessing capitalization, etc.

Reply:
Turns out, we even already have an AutoConfigure Enumeration hanging around elsewhere (thanks to Sean F!) https://github.com/CitrineInformatics/citrine-python/blob/main/src/citrine/builders/auto_configure.py#L28

Reply:
Yes! Please do use that Enum, and move it to a more appropriate location to be used as a standard in the code-base.

I was very annoyed that we had three different ways of matching `FORMULATION` vs `PLAIN` vs etc etc..